### DISCUSSION TITLE
Add email address in Console or Model?

### DISCUSSION THREAD

**Discussion by mcconnon12** [swellstores/community#434] [Help] [answered]
I was wondering if there was something I could add to the Console or Model on Orders to have the email address pass through? Currently the email does not come through. This would make my life so much easier with connecting to my ESP.

**Comment by ericingram** [comment]
How about expanding the account on request?

**Reply by mcconnon12** [reply]
Can you expand on that for me? Thanks.

**Reply by ericingram** [reply]
How are you getting the order in this case? If fetching via console or API, you can expand the account to get the email like this:

```
GET /orders/{id}
{
  "expand": "account"
}
```

**Reply by mcconnon12** [reply]
I would be using the Pabbly Connect Swell App. If I add this in the console would automatically expand to grab the email address on all future orders?

**Reply by ericingram** [reply]
I'm not familiar with the app so I can't say, but in general that's how you would get an email along with an order in the same request.

**Comment by mcconnon12** [comment]
Pretty sure it's just a webhook. Is there something I can add to the model in the Developer section to just relay the email in the initial webhook?

**Reply by ericingram** [reply, marked as answer]
Ah I see. The webhook response just returns the object by itself (an order) and you would have to make a subsequent request to get more data. We've considered adding a configurable `query` to webhooks that would allow you to include more data in the response and this is an interesting data point.

One way to accomplish this would be to create a formula field for the account email on the order model.

```
PUT /:models/orders/fields
{
  "account_email": {
    "type": "string",
    "formula": "account.email"
   }
}
```

However keep in mind that formulas are only executed when an order is created or updated. If the customer changes their email in the meantime, the order would have an old email value until it was updated. Still it should solve your problem with an order create or update webhook.

**Reply by mcconnon12** [reply]
Boom! Just tested this and it seems to be working. This has been a huge thorn in my side... thank you!! Great point... on if they update their email at some point... Is there something similar I can do to capture that as well?

**Reply by ericingram** [reply]
Glad to hear it!

A webhook for `account.updated` would be sent in that case, but not sure how Pabbly handles it

**Reply by mcconnon12** [reply]
Hey Eric, I take it if I wanted to do this for abandoned carts it would be:

PUT /:models/carts/fields
{
  "account_email": {
    "type": "string",
    "formula": "account.email"
   }
}

I appreciate the help.

**Reply by ericingram** [reply]
Yes, that's correct.

Also with Swell Apps in alpha, you could solve this in another way with app function to listen to those events and then send your own payload, with account email or other data, directly to Pabbly. Would be nice to see a Pabbly app in the marketplace at some point of course.

**Reply by mcconnon12** [reply]
Hey Eric I was testing this and the email is not passing it through...  Based on the webhook data it looks like this might be the correct path:

PUT /:models/carts/cart.abandoned/fields

Does that seem right to you?

**Reply by ericingram** [reply]
The API endpoint doesn't look right. If you're just adding a field to the card model it should be 
```
PUT /:models/carts/fields
```

**Reply by mcconnon12** [reply]
Yea, that's what I have but for some reason the email is not coming over... Is there any reason why it might not? Like it's already an existing customer or something like that?

**Reply by ericingram** [reply]
Ah. Since you're referring to the `cart.abandoned` event; webhooks are designed to send the event payload itself which in many cases does not include the entire record. Sometimes it's just an ID reference and relevant changes. What does the payload look like?

**Reply by mcconnon12** [reply]
Hey Eric,

Re-routing here... sorry the delay. I had a few irons in the fire. 

Here's an example of what's coming over... This is for subscriptions but it's similar.

  },
  "req_id": "661298add7e2490012bb1784",
  "date_created": "2024-04-07T12:59:26.287Z",
  "date_updated": "2024-04-07T12:59:26.346Z",
  "webhooks_pending": 2,
  "id": "661298aed7e2490012bb17a0",
  "$type": "subscriptions/subscription.paused"
}

Is there anyway for Swell to include the email in the record? It would solve so many headaches on our end.

I appreciate you taking the time. 

Best,
Shawn

**Reply by ericingram** [reply]
@mcconnon12 we've had a number of people ask about enhancing the values sent by a webhook, but the platform doesn't currently have that functionality. It's expected that the receiver will fetch additional values based on the event.

We might add the ability to specify a query with "includes" for webhooks in the future.

For now I think the best option is to use an App function which is hosted by Swell and can receive webhooks, fetch additional data, and send to any system you need.
